File: client/src/types/family.ts
import { IUser } from "./auth";

export interface IFamily {
  id: number;
  name: string;
  owner_id: number;
  created_at: string;
  updated_at: string;
}

export interface IFamilyResponse {
  family: IFamily;
  members: IUser[];
}

export interface IInviteResponse {
  message: string;
}

File: server/models/family.go
package models

import "time"

type Family struct {
	ID        uint      `gorm:"primaryKey" json:"id"`
	Name      string    `gorm:"not null" json:"name"`
	OwnerID   uint      `json:"owner_id"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}
